File: app/src/main/java/it/cammino/risuscito/database/dao/ListePersDao.java
package it.cammino.risuscito.database.dao;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;

import it.cammino.risuscito.database.entities.ListaPers;

@Dao
public interface ListePersDao {

    @Query("DELETE FROM listapers")
    void truncateTable();

    @Query("SELECT * FROM listapers ORDER BY id ASC")
    LiveData<List<ListaPers>> getLiveAll();

    @Query("SELECT * FROM listapers ORDER BY id ASC")
    List<ListaPers> getAll();

    @Update
    void updateLista(ListaPers lista);

    @Insert
    void insertLista(ListaPers lista);

    @Query("SELECT * FROM listapers WHERE id = :id")
    ListaPers getListById(int id);

    @Delete
    void deleteList(ListaPers lista);

}

File: app/src/main/java/it/cammino/risuscito/database/dao/ListePersDao.kt
package it.cammino.risuscito.database.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import it.cammino.risuscito.database.entities.ListaPers

@Dao
interface ListePersDao {

    @Query("SELECT * FROM listapers ORDER BY id ASC")
    fun liveAll(): LiveData<List<ListaPers>>

    @Query("SELECT * FROM listapers ORDER BY id ASC")
    fun all(): List<ListaPers>

    @Query("DELETE FROM listapers")
    fun truncateTable()

    @Update
    fun updateLista(lista: ListaPers)

    @Insert
    fun insertLista(lista: ListaPers)

    @Query("SELECT * FROM listapers WHERE id = :id")
    fun getListById(id: Int): ListaPers?

    @Query("SELECT * FROM listapers WHERE id = :id")
    fun getLiveListById(id: Int): LiveData<ListaPers>?

    @Delete
    fun deleteList(lista: ListaPers)

}
